File: ios/Wavepoint/Analytics/CrashReporting.swift
protocol CrashReporting: Sendable {
  func record(_ category: AnalyticsErrorCategory)
}

struct NoOpCrashReporting: CrashReporting {
  func record(_: AnalyticsErrorCategory) {}
}

final class CrashReportingClient: CrashReporting, @unchecked Sendable {
  private let recordCategory: (AnalyticsErrorCategory) -> Void

  init(recordCategory: @escaping (AnalyticsErrorCategory) -> Void) {
    self.recordCategory = recordCategory
  }

  func record(_ category: AnalyticsErrorCategory) {
    recordCategory(category)
  }
}

File: android/app/src/main/java/ai/mapier/swipe/analytics/CrashReporting.kt
package ai.mapier.swipe.analytics

fun interface CrashReporting {
  fun record(category: AnalyticsErrorCategory)
}

object NoOpCrashReporting : CrashReporting {
  override fun record(category: AnalyticsErrorCategory) = Unit
}

class CrashReportingClient(
  private val recordCategory: (AnalyticsErrorCategory) -> Unit,
) : CrashReporting {
  override fun record(category: AnalyticsErrorCategory) {
    recordCategory(category)
  }
}
